app/models/mixins/service_mixin.rb

```diff
@@ -64,6 +58,8 @@ def delay_type(action)
   end
 
   def each_group_resource(grp_idx = nil)
+    return enum_for(:each_group_resource) unless block_given?
```

Comment:
Was this change part of the refactor?

Reply:
yes, this allows us to call `each_group_resource.collect`